python/pyarrow/array.pxi

```diff
@@ -2360,6 +2360,25 @@ cdef class StructArray(Array):
         result.validate()
         return result
 
+    def sort_by(self, fieldname, order="ascending"):
```

Comment:
It seems a bit strange to have this method only on StructArray IMO. Also for other arrays, sorting takes some effort with the current API.

Reply:
This is sorting by a given child field. So it wouldn't make sense on other arrays (including unions).

Reply:
But it does make sense to sort just the array. 

What I mean is, also for sorting an array, it takes some code:

```
arr.take(pc.array_sort_indices(arr, order))
```

vs for struct array:

```
arr.take(pc.array_sort_indices(arr.field(name), order)
```

Once you have the first snippet, I don't see why the second snippet would necessarily warrant a helper function (and the first not)

Reply:
Well, we can expose both if we want, though @amol- 's ticket seems to be specifically for `sort_by` :-)

Reply:
I know, but so my feedback is that I am personally -1 on adding a special case method `sort_by`